File: frontend/src/api/school.ts
import { apiClient } from './client'

export interface School {
  school_id: number
  name: string
  address: string
  phone: string
  logo_path: string | null
  bank_name: string
  account_title: string
  account_number: string
  iban: string
  fee_due_day: number
  challan_note: string | null
}

export async function getSchool(): Promise<School> {
  const { data } = await apiClient.get<School>('/school')
  return data
}

export async function updateSchool(payload: Partial<School>): Promise<School> {
  const { data } = await apiClient.put<School>('/school', payload)
  return data
}

export async function uploadLogo(file: File): Promise<School> {
  const formData = new FormData()
  formData.append('file', file)
  const { data } = await apiClient.post<School>('/school/logo', formData, {
    headers: { 'Content-Type': 'multipart/form-data' },
  })
  return data
}

File: backend/app/schemas/school.py
from pydantic import BaseModel, ConfigDict


class SchoolUpdate(BaseModel):
    """Fields editable via PUT /school. logo_path is deliberately excluded —
    it's server-managed exclusively by POST /school/logo, so a settings save
    can never accidentally clobber it (see router for why this matters)."""
    name: str
    address: str
    phone: str
    bank_name: str
    account_title: str
    account_number: str
    iban: str
    fee_due_day: int = 10
    challan_note: str | None = None


class SchoolOut(SchoolUpdate):
    school_id: int
    logo_path: str | None = None

    model_config = ConfigDict(from_attributes=True)
